kornia/augmentation/augmentation.py

```diff
@@ -1911,14 +1911,15 @@ def __init__(
         super(RandomInvert, self).__init__(
             p=p, return_transform=return_transform, same_on_batch=same_on_batch, p_batch=1.0
         )
+        self.max_val = max_val
 
     def __repr__(self) -> str:
         return self.__class__.__name__ + f"({super().__repr__()})"
 
     def apply_transform(
         self, input: torch.Tensor, params: Dict[str, torch.Tensor], transform: Optional[torch.Tensor] = None
     ) -> torch.Tensor:
-        return invert(input)
+        return invert(input, torch.as_tensor(self.max_val, device=input.device, dtype=input.dtype))
```

Comment:
if we assume that `max_val` can be a float we should update typing too.

Reply:
I will update the typing. This was only meant to maintain the devices and dtypes across different tensors.